tests/scripts/thread-cert/Cert_5_2_06_RouterDowngrade.py

```diff
@@ -221,36 +267,107 @@ def test(self):
-        command.check_address_release(msg, self.nodes[LEADER])
-
-        # 4 & 5
         router1_rloc = self.nodes[DUT_ROUTER1].get_ip6_address(config.ADDRESS_TYPE.RLOC)
         self.assertTrue(self.nodes[LEADER].ping(router1_rloc))
 
+    def verify(self, pv):
+        pkts = pv.pkts
+        pv.summary.show()
+
+        LEADER = pv.vars['LEADER']
+        LEADER_RLOC = pv.vars['LEADER_RLOC']
+        ROUTER_1 = pv.vars['ROUTER_1']
+        ROUTER_1_RLOC = pv.vars['ROUTER_1_RLOC']
+
+        # Step 1: Ensure topology is formed correctly
+
+        for i in range(1, 24):
+            pv.verify_attached('ROUTER_%d' % i)
+
+        # Step 3: Allow enough time for for the DUT to get Network Data Updates
+        #         and resign its Router ID.
+        #         The DUT MUST first reconnect to the network as a Child by
+        #         sending properly formatted Parent Request and Child ID Request
+        #         messages.
+        #         Once the DUT attaches as a Child, it MUST send an Address
+        #         Release Message to the Leader:
+        #         CoAP Request URI
+        #             coap://[<leader address>]:MM/a/ar
+        #         CoAP Payload
+        #             MAC Extended Address TLV
+        #             RLOC16 TLV
+
+        pkts.filter_wpan_src64(ROUTER_1).\
+            filter_LLARMA().\
+            filter_mle_cmd(MLE_PARENT_REQUEST).\
+            filter(lambda p: {
+                              CHALLENGE_TLV,
+                              MODE_TLV,
+                              SCAN_MASK_TLV,
+                              VERSION_TLV
+                              } <= set(p.mle.tlv.type) and\
+                   p.ipv6.hlim == 255 and\
+                   p.mle.tlv.scan_mask.r == 1 and\
+                   p.mle.tlv.scan_mask.e == 0
+                  ).\
+            must_next()
+
+        _pkt = pkts.filter_wpan_src64(ROUTER_1).\
+            filter_mle_cmd(MLE_CHILD_ID_REQUEST).\
+            filter(lambda p: {
+                              LINK_LAYER_FRAME_COUNTER_TLV,
+                              MODE_TLV,
+                              RESPONSE_TLV,
+                              TIMEOUT_TLV,
+                              TLV_REQUEST_TLV,
+                              ADDRESS16_TLV,
+                              NETWORK_DATA_TLV,
+                              VERSION_TLV
+                    } <= set(p.mle.tlv.type) and\
+                   p.mle.tlv.addr16 is nullField and\
+                   p.thread_nwd.tlv.type is nullField
+                  ).\
+            must_next()
+        _pkt.must_not_verify(lambda p: (ADDRESS_REGISTRATION_TLV) in p.mle.tlv.type)
+
+        _pkt = pkts.filter_wpan_src64(ROUTER_1).\
+            filter_ipv6_dst(LEADER_RLOC).\
+            filter_coap_request(ADDR_REL_URI).\
+            filter(lambda p: {
+                              NL_MAC_EXTENDED_ADDRESS_TLV,
+                              NL_RLOC16_TLV
```

Comment:
nit: Is it worth to verify that the RLOC16 TLV equals to Router1's RLOC16 ?

Reply:
added the resign rloc16 check